python/graphstorm/gsf.py

```diff
@@ -319,21 +324,38 @@ def create_builtin_lp_model(g, config, train_task):
         if get_rank() == 0:
             print('use dot product for single-etype task.')
             print("Using inner product objective for supervision")
-        decoder = LinkPredictDotDecoder(model.gnn_encoder.out_dims \
-                                            if model.gnn_encoder is not None \
-                                            else model.node_input_encoder.out_dims)
+        if config.lp_edge_weight_for_loss is None:
+            decoder = LinkPredictDotDecoder(model.gnn_encoder.out_dims \
+                                                if model.gnn_encoder is not None \
+                                                else model.node_input_encoder.out_dims)
+        else:
+            decoder = LinkPredictWeightedDotDecoder(model.gnn_encoder.out_dims \
```

Comment:
do we need two classes `LinkPredictDotDecoder` and `LinkPredictWeightedDotDecoder`?

Reply:
Yes, we do.
I don't want to make LinkPredictDotDecoder to complex to use. Basically LinkPredictWeightedDotDecoder inherits LinkPredictDotDecoder and rewrite the `forward` function.